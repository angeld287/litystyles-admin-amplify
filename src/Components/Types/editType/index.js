import React from 'react';
import useEditType from './useEditType';
import { Button, Spinner } from "@blueprintjs/core";

const EditType = () => {
	const { onSubmit, type, error, name, code } = useEditType();

	if (Object.entries(type).length === 0 && type.constructor === Object) return <Spinner />;

	if (error) return <h2>Ha ocurrido un error</h2>;

	return (
		<div align="center" style={{marginTop: 5}}>
				<p className="h4 text-center py-4">Editar Tipo</p>

				<label htmlFor="name" className="grey-text font-weight-light">
					Nombre del Tipo:
				</label>
				<input
					name="name"
					autoComplete="off"
					defaultValue={type.name}
					className="form-control"
					ref={name}
				/>

				<br />

				<label htmlFor="code" className="grey-text font-weight-light">
					Codigo del Tipo:
				</label>
				<input
					name="code"
					autoComplete="off"
					className="form-control"
					defaultValue={type.code}
					ref={code}
				/>

				<br />

				<div className="text-center py-4 mt-3">
					<Button onClick={onSubmit} className="btn btn-outline-blue" >
						Guardar
					</Button>
				</div>
		</div>
	);
};

export default EditType;
