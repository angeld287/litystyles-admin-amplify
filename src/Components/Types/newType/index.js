import React from 'react';
import { Button} from "@blueprintjs/core";
import useNewType from './useNewType';

const NewType = () => {
	const { name, code, onSubmit } = useNewType();

	return (
		<div align="center" style={{marginTop: 5}}>
				<p className="h4 text-center py-4">Nuevo Tipo</p>

				<label htmlFor="name" className="grey-text font-weight-light">
					Nombre del Tipo:
				</label>
				<input
					name="name"
					autoComplete="off"
					className="form-control"
					//value=""
					ref={name}
				/>

				<br />

				<label htmlFor="code" className="grey-text font-weight-light">
					Codigo del Tipo:
				</label>
				<input
					name="code"
					autoComplete="off"
					className="form-control"
					ref={code}
				/>
				<br />

				<div className="text-center py-4 mt-3">
					<Button onClick={onSubmit}>
						Agregar
					</Button>
				</div>
		</div>
	);
};

export default NewType;
