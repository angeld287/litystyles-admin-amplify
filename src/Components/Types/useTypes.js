import React, { useState, useEffect, Fragment } from 'react';
import { API, graphqlOperation } from 'aws-amplify';
import { listTypes } from '../../graphql/queries';
import Swal from 'sweetalert2';
import { deleteType } from '../../graphql/mutations';

const useTypes = () => {
	const [ list, setList ] = useState([]);
	const [ loading, setLoading ] = useState(true);
	const [ error, setError ] = useState(false);

	useEffect(() => {
		let didCancel = false;

		const fetch = async () => {
			var api = [];

			try {
				api = await API.graphql(graphqlOperation(listTypes));
			} catch (error) {
				setLoading(false);
				setError(true);
			}

			if (!didCancel) {
				setList(api.data.listTypes.items);
				setLoading(false);
			}
		};

		fetch();

		return () => {
			didCancel = true;
		};
	}, []);

	const _handleDelete = async (id) => {
		const result = await Swal.fire({
			title: 'Â¿Desea eliminar el tipo?',
			icon: 'warning',
			showCancelButton: true,
			confirmButtonColor: '#3085d6',
			cancelButtonColor: '#d33',
			confirmButtonText: 'Eliminar',
			cancelButtonText: 'Cancelar'
		});

		var input = {
			id
		};

		if (result.value) {
			try {
				await API.graphql(graphqlOperation(deleteType, { input }));
				Swal.fire('Eliminado correctamente!', '', 'success');
				setList(list.filter((e) => e.id !== id));
			} catch (error) {
				Swal.fire('Error', 'Intentelo nuevamente', 'error');
			}
		}
	};

	return { list, error, loading, _handleDelete };
};

export default useTypes;
