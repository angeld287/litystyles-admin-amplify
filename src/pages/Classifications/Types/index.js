import React, { useState, useContext, useEffect, useMemo } from "react";
import { Layout } from 'antd';
import CustomButton from "../../../Components/CustomButton";
import CustomTable from '../../../Components/CustomTable/CustomTable'
import CustomModal from "../../../Components/CustomModal";
import { TypesContext } from "../../../providers";
import { DeleteOutlined, EditOutlined, PlusCircleOutlined } from '@ant-design/icons';
import Swal from "sweetalert2";
import { setModuleStates } from "../../../utils/Items/Utils";

const Types = () => {
    const [show, setShow] = useState(false);
    const [edit, setEdit] = useState(false);
    const [add, setAdd] = useState(false);

    const [id, setId] = useState('');
    const [name, setName] = useState('');
    const [code, setCode] = useState('');
    const [typeItems, setTypeItems] = useState([]);
    const [loading, setLoading] = useState(false);
    const [dlBtnLoading, setDlBtnLoading] = useState('');

    const { Content } = Layout;

    const handleClose = () => setShow(false);

    const { items, addItem, editItem, updateDeleteItem, getItemsNextToken, itemsLoading } = useContext(TypesContext);

    const fields = useMemo(() => [
        { code, setCode },
        { name, setName }
    ], []);

    const openItem = (e) => {
        try {
            setId(e.id);
            setModuleStates(fields, e)
            setEdit(true);
            setAdd(false)
            setShow(true);
        } catch (error) {
            throw new Error('Type - 01: ', error)
        }
    }

    const setType = async () => {
        let result = false;
        if (name === "" || name === null) {
            Swal.fire('Campo Obligatorio', 'Favor completar el campo Nombre', 'error');
            return;
        }

        if (code === "" || code === null) {
            Swal.fire('Campo Obligatorio', 'Favor completar el campo Codigo', 'error');
            return;
        }

        setLoading(true);
        try {
            if (add) {
                result = await addItem({ name: name, code: code });
            } else if (edit) {
                result = await editItem({ id: id, name: name, code: code });
            }
        } catch (e) {
            Swal.fire((add ? 'Creacion' : 'Edicion') + ' de Tipo', 'El proceso de ' + (add ? 'creacion' : 'edicion') + ' ha fallado, intentelo mas tarde', 'error');
        }

        setEdit(false);
        setAdd(false);
        setLoading(false);
        setShow(false);

        if (result !== false) {
            Swal.fire((add ? 'Creacion' : 'Edicion') + ' de Tipo', 'El proceso ha finalizado correctamente', 'success');
        } else {
            Swal.fire((add ? 'Creacion' : 'Edicion') + ' de Tipo', 'El proceso de ' + (add ? 'creacion' : 'edicion') + ' ha fallado, intentelo mas tarde', 'error');
        }
    }

    const alertDeleteItem = async (id) => {
        try {
            const result = await Swal.fire({ title: "Esta seguro que desea eliminar el Tipo?", icon: "warning", showCancelButton: true });
            if (result.isConfirmed && !result.isDenied) {
                await deleteItem(id);
            }
        } catch (error) {
            throw new Error('Type - 02: ', error)
        }
    }

    const deleteItem = async (id) => {
        let result = false;
        setDlBtnLoading(id);

        try {
            await updateDeleteItem(id);
            result = true;
        } catch (error) {
            result = false;
        }

        setDlBtnLoading('');

        if (result === true) {
            Swal.fire('Eliminacion de Tipo', 'El proceso ha finalizado correctamente', 'success');
        } else {
            Swal.fire('Eliminacion de Tipo', 'El proceso de eliminacion ha fallado, intentelo mas tarde', 'error');
        }
    }

    //update the item list when a crud event occurs
    useEffect(() => {
        try {
            if (items !== undefined) {
                setTypeItems(items.map(e => ({
                    nombre: e.name,
                    codigo: e.code,
                    acciones: [
                        { id: e.id, color: 'blue', icon: EditOutlined, onClicAction: () => { openItem(e) }, text: "Editar" },
                        { id: e.id, color: 'red', icon: DeleteOutlined, onClicAction: () => { alertDeleteItem(e.id) }, loading: dlBtnLoading === e.id, text: "Remover" }
                    ],
                    id: e.id
                })))
            }
        } catch (error) {
            throw new Error('Type - 03: ', error)
        }
    }, [items, dlBtnLoading]);

    const inputs = useMemo(() => [
        { label: "Nombre", type: "text", readOnly: (!edit && !add), onChange: e => setName(e.target.value), value: name },
        { label: "Codigo", type: "text", readOnly: (!edit && !add), onChange: e => setCode(e.target.value), value: code },
    ], [show, add, edit, name, code]);

    const _headers = useMemo(() => ['Nombre', 'Codigo', 'Acciones'], []);
    const _TypeItems = useMemo(() => typeItems, [typeItems]);

    return (
        <Content>
            <CustomButton id="typeAddBtn" className="btn-1" style="blue" intent="Primary" onClick={(e) => { e.preventDefault(); setAdd(true); setEdit(false); setShow(true); }} Icon={PlusCircleOutlined}>Agregar Tipo</CustomButton>
            <CustomTable headers={_headers} items={_TypeItems} itemsLoading={itemsLoading} getItemsNextToken={getItemsNextToken} />
            {/* Modal para editar y ver detalle de Tipo */}
            <CustomModal fields={fields} loading={loading} title={edit ? 'Editar Tipo' : add ? 'Agregar Tipo' : 'Ver Tipo'} visible={show} onOk={setType} onCancel={handleClose} inputs={inputs} />
        </Content>
    )
}

export default Types;