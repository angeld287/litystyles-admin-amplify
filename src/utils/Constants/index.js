const SIZE_1024_768 = { name: "_1024_768", dimensions: { width: 1024, height: 768 } }
const SIZE_480_320 = { name: "_480_320", dimensions: { width: 480, height: 320 } }
const SIZE_320_240 = { name: "_320_240", dimensions: { width: 320, height: 240 } }

export const IMAGES = { SIZE_1024_768, SIZE_480_320, SIZE_320_240 };

