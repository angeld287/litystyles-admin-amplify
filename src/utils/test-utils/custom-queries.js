import { queryHelpers, buildQueries } from '@testing-library/react'

// The queryAllByAttribute is a shortcut for attribute-based matchers
// You can also use document.querySelector or a combination of existing
// testing library utilities to find matching nodes for your query
const queryAllByDataCy = (...args) =>
    queryHelpers.queryAllByAttribute('data-cy', ...args)

const queryAllByAriaOwns = (...args) =>
    queryHelpers.queryAllByAttribute('aria-owns', ...args)

const queryAllByClass = (...args) =>
    queryHelpers.queryAllByAttribute('class', ...args)

const queryAllByType = (...args) =>
    queryHelpers.queryAllByAttribute('type', ...args)

const getMultipleError = (c, value) =>
    `Found multiple elements with the attribute: ${value}`
const getMissingError = (c, value) =>
    `Unable to find an element with the attribute: ${value}`

const [
    queryByDataCy,
    getAllByDataCy,
    getByDataCy,
    findAllByDataCy,
    findByDataCy,
] = buildQueries(queryAllByDataCy, getMultipleError, getMissingError)

const [
    queryByClass,
    getAllByClass,
    getByClass,
    findAllByClass,
    findByClass,
] = buildQueries(queryAllByClass, getMultipleError, getMissingError)

const [
    queryByAriaOwns,
    getAllByAriaOwns,
    getByAriaOwns,
    findAllByAriaOwns,
    findByAriaOwns,
] = buildQueries(queryAllByAriaOwns, getMultipleError, getMissingError)

const [
    queryByType,
    getAllByType,
    getByType,
    findAllByType,
    findByType,
] = buildQueries(queryAllByType, getMultipleError, getMissingError)

export {
    queryByDataCy,
    queryAllByDataCy,
    getByDataCy,
    getAllByDataCy,
    findAllByDataCy,
    findByDataCy,
    getAllByAriaOwns,
    queryByAriaOwns,
    getByAriaOwns,
    findAllByAriaOwns,
    findByAriaOwns,
    queryAllByAriaOwns,
    queryByClass,
    getAllByClass,
    getByClass,
    findAllByClass,
    findByClass,
    queryAllByClass,
    queryByType,
    getAllByType,
    getByType,
    findAllByType,
    findByType,
}