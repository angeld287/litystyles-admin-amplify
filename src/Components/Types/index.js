import React, { Fragment } from 'react';
import useTypes from './useTypes';
import { useHistory } from 'react-router-dom';
import { Table, Container, Row, Col, ButtonGroup, Modal, Form } from 'react-bootstrap';
import { Button, Spinner, Icon } from "@blueprintjs/core";

import './index.css';

const Types = (props) => {
	const { loading, error, list, _handleDelete } = useTypes();
	let history = useHistory();

	if(loading) return(<Spinner/>)

	if (error) return <h2>Ha ocurrido un error</h2>;

	const redirect = (path) => {
        window.location.href = path;
	};

	const _list = (list !== null)?([].concat(list)
		.map((item,i) =>
			(
				<tr key={i}>
					<td>{i+1}</td>
					<td>{item.name}</td>
					<td>{item.code}</td>
					<td><Button style={{marginRight: 1}} intent="primary" icon="edit" onClick={e => { e.preventDefault(); history.push('/types/'+item.id+'/edit')}}></Button><Button style={{marginRight: 1}} intent="Danger" icon="delete" onClick={e => { e.preventDefault(); _handleDelete(item.id)}}></Button></td>
				</tr>
			)
		)):(<td></td>)

	return (
		<Container fluid>
			<h3 className="mt-5">Tipos</h3>
			<Row>
				<Col sm={2}><Button loading={false} intent="Primary" onClick={(e) => {e.preventDefault(); redirect('types/new');}} icon="add"></Button></Col>
			</Row>
			<div style={{marginTop:20}}>
				<Table striped bordered hover>
					<thead>
						<tr>
							<th>No.</th>
							<th>Nombre</th>
							<th>Codigo</th>
							<th>Acciones</th>
						</tr>
					</thead>
					<tbody>
						{_list}
					</tbody>
				</Table>
			</div>
		</Container>
	);
};

export default Types;
