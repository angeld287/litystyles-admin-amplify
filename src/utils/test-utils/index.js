// test-utils.js
import { render, queries } from '@testing-library/react'
import * as customQueries from './custom-queries'
import * as customFunctionalities from './functionalities';

const customRender = (ui, options) =>
    render(ui, { queries: { ...queries, ...customQueries, ...customFunctionalities }, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }