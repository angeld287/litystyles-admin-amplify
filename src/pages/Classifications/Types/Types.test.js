import { render, screen, fireEvent, prettyDOM, waitFor } from '../../../utils/test-utils';
import Types from './index';
import MockTypesProvider from './__mocks__/TypesProvider.js';
import MockCategoriesProvider from '../Categories/__mocks__/CategoriesProvider.js';

let component;

beforeEach(() => {
    component = render(
        <MockTypesProvider>
            <MockCategoriesProvider>
                <Types />
            </MockCategoriesProvider>
        </MockTypesProvider>
    )
});

test('Verify correct renderization of Types Component', () => {
    const { asFragment } = render(<Types />);
    expect(asFragment()).toMatchSnapshot();
})

test('Add Types without Name', async () => {
    const { getAllByClass } = component;

    //clic "Agregar Tipo" button
    fireEvent.click(screen.getByText(/Agregar Tipo/i));

    //clic "Guardar" button
    fireEvent.click(screen.getByText(/Guardar/i));

    await waitFor(() => {
        //verify if validation message is shown
        expect(screen.getByText("Favor completar el campo Nombre")).toBeInTheDocument();

        //clic to ok button
        fireEvent.click(getAllByClass(/swal2-confirm/i)[0])
    });

})

test('Add Types without Codigo', async () => {
    const { getAllByClass, writeInInputFoundByPlaceHolder } = component;

    //clic "Agregar Tipo" button
    fireEvent.click(screen.getByText(/Agregar Tipo/i));

    writeInInputFoundByPlaceHolder(/Nombre/i, "Tipo Prueba");

    //clic "Guardar" button
    fireEvent.click(screen.getByText(/Guardar/i));

    await waitFor(() => {
        //verify if validation message is shown
        expect(screen.getByText("Favor completar el campo Codigo")).toBeInTheDocument();

        //clic to ok button
        fireEvent.click(getAllByClass(/swal2-confirm/i)[0])
    });

})

test('Add Types to the List', async () => {

    const { rerender, getAllByClass, writeInInputFoundByPlaceHolder } = component;

    //Get the number of existing items in the table
    let tableRows = getAllByClass(/ant-table-row/i);

    //clic "Agregar Tipo" button
    fireEvent.click(screen.getByText(/Agregar Tipo/i));

    //Complete the field "Nombre"
    writeInInputFoundByPlaceHolder(/Nombre/i, "Tipo Prueba");

    //Complete the field "Codigo"
    writeInInputFoundByPlaceHolder(/Codigo/i, "100");

    //clic "Guardar" button
    fireEvent.click(screen.getByText(/Guardar/i));

    await waitFor(() => {

        expect(getAllByClass(/swal2-success-ring/)[0]).toBeInTheDocument();
        expect(screen.getByText(/El proceso ha finalizado correctamente/)).toBeInTheDocument();

        //clic to ok button
        fireEvent.click(getAllByClass(/swal2-confirm/i)[0])
    });

    //rerender the componet to reflect the new item added to the table
    rerender(<MockTypesProvider><Types /></MockTypesProvider>);


    //Verify that the item has been created and inserted in the table
    expect(getAllByClass(/ant-table-row/i).length).toBeGreaterThan(tableRows.length);

});