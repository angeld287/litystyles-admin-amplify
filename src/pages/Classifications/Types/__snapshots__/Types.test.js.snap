// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify correct renderization of Types Component 1`] = `
<DocumentFragment>
  <main
    class="ant-layout-content"
  >
    <button
      class="ant-btn btn-1"
      type="button"
    >
      <span
        aria-label="plus-circle"
        class="anticon anticon-plus-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="plus-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
          />
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
        </svg>
      </span>
      <span>
        Agregar Tipo
      </span>
    </button>
    <div
      style="margin-top: 20px;"
    >
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table ant-table-empty"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Nombre
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Codigo
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Acciones
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-placeholder"
                      >
                        <td
                          class="ant-table-cell"
                          colspan="3"
                        >
                          <div
                            class="ant-empty ant-empty-normal"
                          >
                            <div
                              class="ant-empty-image"
                            >
                              <svg
                                class="ant-empty-img-simple"
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <g
                                  fill="none"
                                  fill-rule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    class="ant-empty-img-simple-ellipse"
                                    cx="32"
                                    cy="33"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    class="ant-empty-img-simple-g"
                                    fill-rule="nonzero"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      class="ant-empty-img-simple-path"
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <div
                              class="ant-empty-description"
                            >
                              No Data
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
